src/main/java/org/truffleruby/platform/graal/AssertConstantNode.java

```diff
@@ -31,7 +31,7 @@ public Object assertCompilationConstant(Object value) {
 
         if (!compilationConstant[0]) {
             CompilerDirectives.transferToInterpreterAndInvalidate();
-            throw new RaiseException(coreExceptions().internalErrorAssertConstantNotConstant(this));
+            throw new RaiseException(coreExceptions().internalErrorAssertConstantNotConstant(this), true);
```

Comment:
These should be something like Truffle::OptimizationException or so, they are internal and separate and only used for testing.

Reply:
`Truffle::CompilationException` seems a better name.

Reply:
I used `Truffle::GraalError`.